rust/windows-client/src-tauri/src/client/updates.rs

```diff
@@ -0,0 +1,132 @@
+//! Module to check the Github repo for new releases
+
+use crate::client::about::get_cargo_version;
+use std::str::FromStr;
+use url::Url;
+
+#[derive(Debug, serde::Deserialize)]
+pub(crate) struct Release {
+    /// e.g. <https://github.com/firezone/firezone/releases/tag/1.0.0-pre.8>
+    pub html_url: Url,
+    /// e.g. 1.0.0-pre.8
+    pub tag_name: semver::Version,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum Error {
+    #[error("Non-OK HTTP status")]
+    HttpStatus(reqwest::StatusCode),
+    #[error(transparent)]
+    JsonParse(#[from] serde_json::Error),
+    #[error("Our own semver in the exe is invalid, this should be impossible")]
+    OurVersionIsInvalid(semver::Error),
+    #[error(transparent)]
+    Request(#[from] reqwest::Error),
+}
+
+const LATEST_RELEASE_API_URL: &str =
+    "https://api.github.com/repos/firezone/firezone/releases/latest";
+
+/// <https://docs.github.com/en/rest/about-the-rest-api/api-versions?apiVersion=2022-11-28>
+const GITHUB_API_VERSION: &str = "2022-11-28";
+
+/// <https://docs.github.com/en/rest/using-the-rest-api/getting-started-with-the-rest-api?apiVersion=2022-11-28#user-agent-required>
+const USER_AGENT: &str = "Firezone Windows Client";
+
+/// Returns the latest release, even if ours is already newer
+pub(crate) async fn check() -> Result<Release, Error> {
+    let client = reqwest::Client::builder().build()?;
+
+    // Reqwest follows up to 10 redirects by default
+    // https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.redirect
+    // https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#follow-redirects
+    let response = client
+        .get(LATEST_RELEASE_API_URL)
+        .header("User-Agent", USER_AGENT)
```

Comment:
Probably not a biggie, but it might be better to use a more realistic user agent string

```suggestion
        .header("User-Agent", format!("Firezone/{:?} (Windows; Win64; x64)", current_version())
```

Reply:
Just thinking in case this comes up in support requests w/ GitHub

Reply:
Good idea, that way if a specific version is buggy they can block that version alone.

Reply:
I'll still have it say "Client" though, in case the same system runs a gateway in a separate process in the future